File: OhHell/src/main/java/game/play/Card.java
package game.play;

import static org.junit.Assert.assertTrue;

public class Card {

	private Suit suit;
	private int denom;

	public Card(int denom, Suit suit)
	{
		assertTrue( denom > 0 && denom <= 13 );
		this.denom = denom;
		this.suit = suit;
	}

	public Suit getSuit() {
		return suit;
	}

	public void setSuit(Suit suit) {
		this.suit = suit;
	}

	public int getDenom() {
		return denom;
	}

	public void setDenom(int denom) {
		assertTrue( denom > 0 && denom <= 13 );
		this.denom = denom;
	}
	
	public String toString()
	{
		return "" + getDenomName() + " of " + suit.getAbbr();
	}

	private String getDenomName() {
		switch(denom)
		{
		case(13):
			return "Ace";
		case(1):
			return "Two";
		case(2):
			return "Three";
		case(3):
			return "Four";
		case(4):
			return "Five";
		case(5):
			return "Six";
		case(6):
			return "Seven";
		case(7):
			return "Eight";
		case(8):
			return "Nine";
		case(9):
			return "Ten";
		case(10):
			return "Jack";
		case(11):
			return "Queen";
		case(12):
			return "King";
		default:
			System.err.println("Card with no denomination!");
			return "NOTHING!";
		}
	}

	/*
	 * given another card, and knowledge of the suit of the lead, and the current trump,
	 * this method returns true if this card beats the other one, and false if it doesnt.
	 * TODO: UNIT TESTS!!!
	 */
	public boolean beats(Card other, Suit lead, Suit trump) {
		if ( other == null ) // no contender (ie, you're the first card being considered)
		{
			return true;
		}
		if (other.getSuit().equals(trump))
		{
			if ( this.getSuit().equals(trump))
			{
				return this.denom > other.getDenom();
			}
			else
			{
				return false;
			}
		}
		else
		{
			if ( this.getSuit().equals(trump))
			{
				return true;
			}
			else
			{
				if ( other.getSuit().equals(lead))
				{
					if ( this.getSuit().equals(lead))
					{
						return this.denom > other.getDenom();
					}
					else
					{
						return false;
					}
				}
				else
				{
					if ( this.getSuit().equals(lead))
					{
						return true;
					}
					else
					{
						return this.denom > other.getDenom();
					}
				}
			}
		}
		
	}
	
	
}

File: Assets/Scripts/Card.cs
using UnityEngine;

public enum Suit
{
    Clubs, Diamonds, Hearts, Spades
}

// extensions class to provide a .getAbbr() method on the Suit enum
public static class Extensions
{
    public static string GetAbbr(this Suit suit)
    {
        return suit.ToString().Substring(0, 1);
    }
}

public class Card
{
    public int denom;
    public Suit suit;

    public Card(Suit suit, int denom)
    {
        this.denom = denom;
        this.suit = suit;
        if (denom < 2 || denom > 14)
        {
            throw new UnityException("Can't have a card with denomination " + denom.ToString());
        }
    }

    public string GetName()
    {
        return GetDenomName() + " of " + suit.ToString();
    }

    public string GetAbbr()
    {
        return (denom == 14 ? "A" : denom.ToString()) + suit.GetAbbr();
    }

    private string GetDenomName()
    {
        switch (denom)
        {
            case (14):
                return "Ace";
            case (2):
                return "Two";
            case (3):
                return "Three";
            case (4):
                return "Four";
            case (5):
                return "Five";
            case (6):
                return "Six";
            case (7):
                return "Seven";
            case (8):
                return "Eight";
            case (9):
                return "Nine";
            case (10):
                return "Ten";
            case (11):
                return "Jack";
            case (12):
                return "Queen";
            case (13):
                return "King";
            default:
                return "NOTHING!";
        }
    }
}